xcodeproj/internal/xcodeprojinfo.bzl

```diff
@@ -293,8 +293,26 @@ def _skip_target(
         # `bundle_name` can be used to name the bundle instead of the
         # target name. Because of that we use `ctx.rule.attr.generator_name`
         # here to ensure this is always a real target label.
+        label_name = ctx.rule.attr.generator_name
+
+        # If `generator_function` is not `ios_unit_test` it means this target
+        # is wrapped in a macro and `generator_name` does not hold the value
+        # we want. In that case, following rules_apple's naming pattern, there
+        # are two options:
+        #
+        # 1. `ctx.rule.attr.runner` is rules_apple's default and we can use
+        # `ctx.rule.attr.name` as is
+        # 2. `ctx.rule.attr.runner` is custom and `ctx.rule.attr.name` has
+        # the pattern "{target_name}_{runner_name}"
+        #
+        # In both scenarios if we remove the "_{runner_name}" suffix we
+        # get the label_name that we need.
+        if ctx.rule.attr.generator_function != "ios_unit_test":
```

Comment:
Need to check if this is generic enough, will test with other test specific rule kinds

Reply:
`runners` is at the `ios_*_test_suite` level when this code is running we're always dealing with the underlying targets which contain `rules_apple`'s internal test bundle as a dep and `runner` set (to default or a custom one defined by the user).

When a test is wrapped in macro called `macro_foo` then `generator_function` value will be `macro_foo` and not `ios_unit_test`. Which makes me think what we want here is 
```py
ctx.rule.attr.generator_function in ['ios_unit_test', ... <other known test kinds>]
```

Does that addresses your Q?

Reply:
And that's mostly my round about suggestion, being explicit here about the `rules_apple` things we're trying to handle vs. a `!=` test since it'll be quite generous. And mostly thinking about things like `ios_build_test` and so on like you mention.

Reply:
Let's continue on the other thread: https://github.com/MobileNativeFoundation/rules_xcodeproj/pull/2291/files#r1232615504

(ps: `ios_build_test` and similar are not hit here since they're skipped earlier while targets are being processed, so this should be safely running only for `*_(unit|ui)_test` rules only as all of them respect the patterns I mentioned in the comment added to the code above)